eip-0034.md

```diff
@@ -0,0 +1,121 @@
+
+### Collection standard version
+Collection standard version must be present in `R4` as `Int`. This current standard version is `1`. This will enable future radical changes depending on the state of the NFT ecosystem.
+
+### Collection Info
+Collection info must be present in `R5` of the issuance box as `Coll[Byte]`.
+The following JSON contains the structure of collection info.
+```json
+{
+    "name": "NAME_OF_THE_COLLECTION",
+    "description": "DESCRIPTION_OF_THE_COLLECTION",
+    "logo": "URL_TO_COLLECTION_LOGO",
+    "featured_image": "URL_TO_COLLECTION_FEATURED_IMAGE",
+    "banner_image": "URL_TO_COLLECTION_BANNER_IMAGE",
+    "category": "COLLECTION_CATEGORY",
+    "explicit_content": true or false
+}
+```
+
+### Socials
+Socials must be present in `R6` of the issuance box as `Coll[Byte]`.
+The following JSON contains information about the socials of the collection and/or the artist(s).
+```json
+{
+    "twitter": "TWITTER_PROFILE",
+    "instagram": "INSTAGRAM_RPOFILE"
+    ...
+}
+```
+
+## Traits
+Traits information must be present in `R7` of the issuance box as `Coll[Byte]`.
+There are three types of traits:
+- Properties: These are textual traits such as specifying `sex` as `male`.
+- Levels: These are numerical traits that encode some sort of level-like information such as specifying `speed` as `60 out of 100`.
+- Stats: These are numerical traits that encode any numerical information about the NFT such as specifying `age` as `25 out of 50`.
+
+Although levels and stats are very similar, they can be considered different in marketplaces both for UI/UX purposes and also for filtering purposes.
+
+The following specifies the structure for traits.
+```json
+{
+    "properties": ["NAME_OF_PROPERTY1", "NAME_OF_PROPERTY2"],
+    "levels": [
+        {
+            "NAME_OF_LEVEL1": max_alowed_level_number1
+        },
+        {
+            "NAME_OF_LEVEL2": max_alowed_level_number2
+        }
+    ],
+    "stats": [
+        {
+            "NAME_OF_STAT1": max_alowed_stat_number1
+        },
+        {
+            "NAME_OF_STAT2": max_alowed_stat_number2
+        },
+        {
+            "NAME_OF_STAT3": max_alowed_stat_number3
+        }
+    ]
+}
+```
+
+The following is an example of traits for some imaginary NFT.
+```json
+{
+    "properties": ["sex", "hair color", "eye color"],
+    "levels": [
+        {
+            "power": 100 # this means the maximum possible power is 100
+        },
+        {
+            "speed": 180 # this means the maximum possible power is 180
+        }
+    ],
+    "stats": [
+        {
+            "rank": 50 # this means the maximum possible rank is 50
+        }
+        {
+            "number of swords": 1000 # this means the maximum possible number of swords is 1000
+        }
+    ]
+}
+```
+It must be noted that the actual value for traits will be present in the NFT data (refer to [artwork standard]()) and the above numbers for levels and stats only specify the maximum range.
```

Comment:
Might be helpful to add example NFT metadata matching the example given? 

```json
{
	"properties": [
		{
			"sex": "male"
		},
		{
			"hair color": "brown"
		},
		{
			"eye color": "green"
		}
	],
	"levels": [
		{
			"power": 50
		},
		{
			"speed": 90
		}
	],
	"stats": [
		{
			"rank": 11
		},
		{
			"number of swords": 948
		}
	]
}

Reply:
This part has changed altogether. Now the traits are only present in the NFT metadata and not in the collection. After some back and forth, this seems a more reasonable design. Please let me know if this is resolved.

Reply:
so to clarify, the traits section will be in the issuer box or the issuance box?
some of us are (mgpai, cheese, and myself) are looking to use GetBlok.io's Plasma library for manipulating AVL trees in order to scale the nft minting/selling to 10k/100k/1M+ NFTs. Unfortunately, the ues of traits in the issuer box that differ from NFT to NFT complicates this process and would seem to make more sense if it were part of the NFT issuance box itself since traits are intuitively associated to the actual existing object (i.e. the NFT itself/issuance box) and not the precursor object that gave birth to it (i.e. the issuer box).

Reply:
Furthermore, to clarify again as well, the traits are only max/min values or specific values? If they are max/min values, then they would be the same for all NFTs in a collection or all NFTs with the same rarity for example?

Reply:
The problem encountered with this implementation of traits and using Plasma is that the NFT meta-data to be placed in the issuance box is stored in the avl tree. If issuer boxes are the same for all NFTs in a collection or NFTs of the same "rarity" then one could reuse these contracts and issuer boxes in subsequent minting txs.